design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,480 @@
+}
+
+type VolumeMigrationPhase string
+
+const (
+	VolumeMigrationPhasePending    = "Pending"
+	VolumeMigrationPhaseRunning    = "Running"
+	VolumeMigrationPhaseScheduling = "Scheduling"
+	VolumeMigrationPhaseSucceeded  = "Succeeded"
+	VolumeMigrationPhaseFailed     = "Failed"
+	VolumeMigrationPhaseUnknown    = "Unknown"
+)
+
+type VolumeMigrationPhaseTransitionTimestamp struct {
+	Phase                    VolumeMigrationPhase `json:"phase,omitempty"`
+	PhaseTransitionTimestamp metav1.Time          `json:"phaseTransitionTimestamp,omitempty"`
+}
+
+type MigratedVolumeState string
+
+const (
+        MigratedVolumeStateValid    = "Valid"
+        MigratedVolumeStatePending  = "Pending"
+        MigratedVolumeStateRejected = "Rejected"
+)
+
+type VolumeMigrationState struct {
+        MigratedVolume `json:",inline"`
+        State     MigratedVolumeState `json:"state,omitempty"`
+        Reason    *string             `json:"reason,omitempty"`
+}
+
+type VolumeMigrationStatus struct {
+        VolumeMigrationStates       []VolumeMigrationState                    `json:"volumeMigrationStates,omitempty"`
+        VirtualMachineInstanceName  *string                                   `json:"virtualMachineInstanceName,omitempty"`
+        VirtualMachineMigrationName *string                                    json:"virtualMachineMigrationName,omitempty"`
+        StartTimestamp              *metav1.Time                              `json:"startTimestamp,omitempty"`
+        EndTimestamp                *metav1.Time                              `json:"endTimestamp,omitempty"`
+        Phase                       VolumeMigrationPhase                      `json:"phase,omitempty"`
+        PhaseTransitionTimestamps   []VolumeMigrationPhaseTransitionTimestamp `json:"phaseTransitionTimestamps,omitempty"`
+}
+```
+
+Users may define more than one migrated volume, and those volumes may be associated with the same VMI, a different VMI, or none at all. The volumes defined on the same VMI will be categorized and grouped by the volume migration controller.
+
+At the moment, PVCs can only be migrated using this API if they belong to a single running virtual machine. Consequently, all volume migration objects that have migrated volumes that are associated with different VMIs will be rejected by the controller. The object is in the `Pending` state if none of the migrated volumes have been assigned to a VMI, since the VMI might be created at a later time.
+
+The `VolumeMigration` expects a list of source and destination PVCs and the destination PVCs needs to be created either manually by the user or other tools.
+
+According to the `SourcePvcReclaimPolicy`, KubeVirt will determine how to handle the source PVCs. The source PVC will be automatically deleted under the `Delete` policy, but it will remain untouched under the `Retain` policy.
+If the user doesn't specify any reclaim policy, the `Delete` policy will be set as default.
```

Comment:
I don't know if `Delete` should be the default

Reply:
I see advantages and disadvantages. The PVCs are automatically cleaned up if you don't specify anything, but keeping them around could be a more conservative choice.
But neither @vasiliy-ul  or I had a strong opinion about the default value, hence I can set it to `Retain`

Reply:
See https://github.com/kubevirt/community/pull/245#discussion_r1461533893

Reply:
To me it's better safe than sorry with defaults